src/internal/packager/git/push.go

```diff
@@ -32,7 +32,7 @@ func (g *Git) PushRepo(localPath string) error {
 	}
 
 	if err := g.push(repo, spinner); err != nil {
-		spinner.Warnf("Unable to push the git repo %s", basename)
+		spinner.Warnf("Unable to push the git repo %s because of %w. Retrying the push.", basename, err)
```

Comment:
@Yash-C-330 apologies for the delay on this as we work through some other issues in the background, I think the overall flow is correct now but the error unwrapping doesn't print as expected (see below):

![image](https://user-images.githubusercontent.com/3977569/218605955-984351e1-7f22-4457-8a17-19ae625b4698.png)

The following would unwrap this correctly:

```suggestion
		spinner.Warnf("Unable to push the git repo %s (%s). Retrying....", basename, err.Error())
```

Reply:
(here is an example of what that above suggestion outputs)
![image](https://user-images.githubusercontent.com/3977569/218606477-66ff3b77-2b41-4efe-8966-3b1af21ee40d.png)